Realm/RLMApp.mm

```diff
@@ -91,31 +95,18 @@ - (instancetype)initWithBaseURL:(nullable NSString *)baseURL
                 localAppVersion:(nullable NSString *)localAppVersion
         defaultRequestTimeoutMS:(NSUInteger)defaultRequestTimeoutMS {
     if (self = [super init]) {
-        _config.transport_generator = []{
-            return std::make_unique<CocoaNetworkTransport>([RLMNetworkTransport new]);
-        };
-
-        if (baseURL) {
-            std::string base_url;
-            RLMNSStringToStdString(base_url, baseURL);
-            _config.base_url = base_url;
-        }
-        if (transport) {
-            _config.transport_generator = [transport]{
-                return std::make_unique<CocoaNetworkTransport>(transport);
-            };
-        }
-        if (localAppName) {
-            std::string local_app_name;
-            RLMNSStringToStdString(local_app_name, localAppName);
-            _config.local_app_name = local_app_name;
-        }
-        if (localAppVersion) {
-            std::string local_app_version;
-            RLMNSStringToStdString(local_app_version, localAppName);
-            _config.local_app_version = local_app_version;
-        }
-        _config.default_request_timeout_ms = (uint64_t)defaultRequestTimeoutMS;
+        self.baseURL = baseURL;
+        self.transport = transport;
+        self.localAppName = localAppName;
+        self.localAppVersion = localAppVersion;
+        self.defaultRequestTimeoutMS = defaultRequestTimeoutMS;
+        
+        _config.platform = "Realm Cocoa";
+        
+        NSOperatingSystemVersion operatingSystemVersion = [[NSProcessInfo processInfo] operatingSystemVersion];
+        RLMNSStringToStdString(_config.platform_version,
+                               [[NSString alloc] initWithFormat:@"%d.%d.%d", (int)operatingSystemVersion.majorVersion, (int)operatingSystemVersion.minorVersion, (int)operatingSystemVersion.patchVersion]);
```

Comment:
`_config.platform_version = realm::util::format("%1.%2.%2", operatingSystemVersion.majorVersion, operatingSystemVersion.minorVersion, operatingSystemVersion.patchVersion)` skips the NSString conversion.